js/apps/admin-ui/src/identity-providers/add/DiscoverySettings.tsx

```diff
@@ -50,152 +33,92 @@ const Fields = ({ readOnly }: DiscoverySettingsProps) => {
 
   return (
     <div className="pf-c-form pf-m-horizontal">
-      <FormGroup
+      <TextControl
+        name="config.authorizationUrl"
         label={t("authorizationUrl")}
-        fieldId="kc-authorization-url"
-        isRequired
-        validated={
-          errors.config?.authorizationUrl
-            ? ValidatedOptions.error
-            : ValidatedOptions.default
-        }
-        helperTextInvalid={t("required")}
-      >
-        <KeycloakTextInput
-          type="url"
-          data-testid="authorizationUrl"
-          id="kc-authorization-url"
-          validated={
-            errors.config?.authorizationUrl
-              ? ValidatedOptions.error
-              : ValidatedOptions.default
-          }
-          isReadOnly={readOnly}
-          {...register("config.authorizationUrl", { required: true })}
-        />
-      </FormGroup>
-
-      <FormGroup
+        type="url"
+        readOnly={readOnly}
+        rules={{
+          required: t("required"),
+        }}
+      />
+      <TextControl
+        name="config.tokenUrl"
         label={t("tokenUrl")}
-        fieldId="tokenUrl"
-        isRequired
-        validated={
-          errors.config?.tokenUrl
-            ? ValidatedOptions.error
-            : ValidatedOptions.default
-        }
-        helperTextInvalid={t("required")}
-      >
-        <KeycloakTextInput
-          type="url"
-          id="tokenUrl"
-          data-testid="tokenUrl"
-          validated={
-            errors.config?.tokenUrl
-              ? ValidatedOptions.error
-              : ValidatedOptions.default
-          }
-          isReadOnly={readOnly}
-          {...register("config.tokenUrl", { required: true })}
-        />
-      </FormGroup>
-      <TextField
```

Comment:
I forgot we had this component, we should make sure to remove this and replace it with the new fields everywhere.

Reply:
Note not in this PR, just a reminder.

Reply:
also put that in the exelsheet